Flaky test

Our CI often fails, as a false negative, on the test `python -m pytest --count=1000 -nauto openff/toolkit/tests/test_toolkits.py::TestAmberToolsToolkitWrapper::test_assign_fractional_bond_orders_openeye_installed`. This is annoying.

The underlying issue is comparing two lists of floats with `==`, which Pytest sensibly interprets as strict equality:

```shell
(openff-dev) [~] cat test_foo.py && pytest test_foo.py                                                   14:11:57
def test_foo():
    assert 1.0 == 1.0 + 1e-8
=============================================== test session starts ===============================================
platform darwin -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/mattthompson
plugins: xdist-2.5.0, forked-1.4.0, rerunfailures-10.2, nbval-0.9.6, repeat-0.9.1, cov-3.0.0
collected 1 item

test_foo.py F                                                                                               [100%]

==================================================== FAILURES =====================================================
____________________________________________________ test_foo _____________________________________________________

    def test_foo():
>       assert 1.0 == 1.0 + 1e-8
E       assert 1.0 == (1.0 + 1e-08)

test_foo.py:2: AssertionError
============================================= short test summary info =============================================
FAILED test_foo.py::test_foo - assert 1.0 == (1.0 + 1e-08)
================================================ 1 failed in 0.14s ================================================
```

The fractional bond orders are reported from underlying toolkits to something like 6 or more digits of precision, so spooky run-to-run variance is reported as a failure. Of course there's a way around this: [`pytest.approx`](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=approx#pytest-approx).

--- a/openff/toolkit/tests/test_toolkits.py
+++ b/openff/toolkit/tests/test_toolkits.py
@@ -3768,7 +3768,7 @@ class TestAmberToolsToolkitWrapper:
         without_oe = [b.fractional_bond_order for b in mol.bonds]
         GLOBAL_TOOLKIT_REGISTRY.register_toolkit(OpenEyeToolkitWrapper)
 
-        assert with_oe == without_oe
+        assert with_oe == pytest.approx(without_oe, abs=1e-5)
 
 
 class TestBuiltInToolkitWrapper:

